app/Community/Actions/BuildActivePlayersAction.php

```diff
@@ -0,0 +1,175 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Community\Actions;
+
+use App\Community\Data\ActivePlayerData;
+use App\Data\PaginatedData;
+use App\Models\Game;
+use App\Models\User;
+use Illuminate\Pagination\LengthAwarePaginator;
+use Illuminate\Support\Collection;
+use Illuminate\Support\Str;
+
+class BuildActivePlayersAction
+{
+    public function execute(
+        int $perPage = 300,
```

Comment:
The current component in prod initially shows 20 results, then loads all N after the user scrolls.

That behavior isn't really great from a UI performance perspective. The browser freezes on low-power devices because it's a ton of nodes being inserted simultaneously into the DOM.

The new component will either infinite scroll or virtual scroll - I haven't decided yet. This action will support either approach.